src/h5web/visualizations/index.test.ts

```diff
@@ -1,559 +1,408 @@
-import {
-  HDF5Collection,
-  HDF5TypeClass,
-  HDF5ShapeClass,
-  HDF5Metadata,
-  HDF5Entity,
-  HDF5Attribute,
-  HDF5LinkClass,
-} from '../providers/models';
+import { HDF5Metadata, HDF5Entity } from '../providers/models';
 import { Vis, VIS_DEFS } from '.';
 import { NxInterpretation } from './nexus/models';
-
-const baseGroup = { id: 'group', collection: HDF5Collection.Groups as const };
-const baseDataset = {
-  id: 'dataset',
-  collection: HDF5Collection.Datasets as const,
-};
-const baseDatatype = {
-  id: 'datatype',
-  collection: HDF5Collection.Datatypes as const,
-};
-
-const nxDataGroup = {
-  ...baseGroup,
-  attributes: [
-    { name: 'NX_class', value: 'NXdata' },
-    { name: 'signal', value: 'my_signal' },
-  ] as HDF5Attribute[],
-  links: [
-    {
-      class: HDF5LinkClass.Hard as const,
-      title: 'my_signal',
-      collection: HDF5Collection.Datasets,
-      id: baseDataset.id,
-    },
-  ],
-};
-
-function supportsEntity(
-  vis: Vis,
-  entity: HDF5Entity,
-  metadata = {} as HDF5Metadata
-): boolean {
-  return VIS_DEFS[vis].supportsEntity(entity, metadata);
+import {
+  makeSimpleDataset,
+  intType,
+  makeGroup,
+  makeDatatype,
+  advancedType,
+  makeDataset,
+  scalarShape,
+  makeSimpleShape,
+  stringType,
+  makeStrAttr,
+  makeMetadata,
+  withAttributes,
+  makeDatasetHardLink,
+  makeNxDataGroup,
+} from '../providers/mock/data';
+
+const datasetIntScalar = makeDataset('dataset_int', intType, scalarShape);
+const datasetFltScalar = makeDataset('dataset_flt', intType, scalarShape);
+const datasetInt0D = makeSimpleDataset('dataset_int_0d', intType, []);
+const datasetInt1D = makeSimpleDataset('dataset_int_1d', intType, [5]);
+const datasetInt2D = makeSimpleDataset('dataset_int_2d', intType, [5, 3]);
+const datasetStr1D = makeSimpleDataset('dataset_str_1d', stringType, [5]);
+const datasetStr2D = makeSimpleDataset('dataset_str_2d', stringType, [5, 3]);
+const datasetFlt3D = makeSimpleDataset('dataset_flt_3d', intType, [5, 3, 1]);
+
+const groupEmpty = makeGroup('group_empty');
```

Comment:
Nitpicking but I would find the variables easier to read if the adjectives were put before: `emptyGroup`, `advancedDatatype`.

The dataset names above are fine to me though.

Reply:
Yeah, I hesitated. I felt like having the most important aspect on the left was actually easier to read where the variables are being initialised. It also fits with the order of the function call and parameters: dataset then type then shape.